Doc/c-api/dict.rst

```diff
@@ -98,9 +98,11 @@ Dictionary Objects
    Return the object from dictionary *p* which has a key *key*.  Return ``NULL``
    if the key *key* is not present, but *without* setting an exception.
 
-   Note that exceptions which occur while calling :meth:`__hash__` and
-   :meth:`__eq__` methods will get suppressed.
-   To get error reporting use :c:func:`PyDict_GetItemWithError()` instead.
+   .. note::
```

Comment:
Since this is providing a caution/warning of a problem with the function's current behavior (and recommending an alternative), I would consider a more appropriate type of admonition here, such as 

```suggestion
   .. caution::
```

or potentially even

```suggestion
   .. warning::
```